guides/keras_nlp/upload.py

```diff
@@ -0,0 +1,167 @@
+"""
+Title: Model Upload
+Author: [Samaneh Saadat](https://github.com/SamanehSaadat)
+Date created: 2024/04/22
+Last modified: 2024/04/22
+Description: An introduction on how to upload a fine-tuned model to model hubs.
+Accelerator: GPU
+"""
+
+"""
+# Introduction
+
+Fine-tuning a machine learning model can yield impressive results for specific tasks.
+Uploading your fine-tuned model to a model hub allow you to share it with the broader community.
+By sharing your models, you'll enhance accessibility for other researchers and developers,
+making your contributions an integral part of the machine learning landscape.
+This can also streamline the integration of your model into real-world applications.
+
+This guide walks you through how to upload your fine-tuned models to popular model hubs such as
+[Kaggle Hub](https://www.kaggle.com/models) and [Hugging Face Hub](https://huggingface.co/models).
+"""
+
+"""
+# Setup
+
+Let's start by installing and importing all the libraries we need. We use KerasNLP for this guide.
+"""
+
+"""shell
+pip install -q --upgrade keras-nlp
+pip install -q --upgrade keras  # Upgrade to Keras 3.
+"""
+
+import os
+
+os.environ["KERAS_BACKEND"] = "jax"
+
+import keras_nlp
+import keras
+
+"""
+# Task Upload
+
+A KerasNLP `Task`, wraps a `keras_nlp.models.Backbone` and a `keras_nlp.models.Preprocessor` to create
+a model that can be directly used for training, fine-tuning, and prediction for a given text problem.
+In this section, we explain how to create a `Task`, fine-tune and upload it to a model hub.
+"""
+
+"""
+## Load Model
+
+If you want to build a Causal LM based on a base model, simply call `keras_nlp.models.Classifier.from_preset`
+and pass a built-in preset identifier.
+"""
+
+causal_lm = keras_nlp.models.CausalLM.from_preset("gpt2_base_en")
+
+
+"""
+## Fine-tune Model
+After loading the model, you can call `.fit()` on the model to fine-tune it.
+"""
+
+# A toy dataset.
+data = ["The quick brown fox jumped.", "I forgot my homework."]
+
+# Fine-tune the `causal_lm` model.
+causal_lm.fit(data, batch_size=2)
+
+"""
+## Save the Model Locally
+
+To upload a model, you need to first save the model locally using `save_to_preset`.
+"""
+
+preset_dir = "./finetuned_gpt2"
+causal_lm.save_to_preset(preset_dir)
+
+"""
```

Comment:
Before showing upload, maybe `os.listdir()` to show this is a directory of config and weights. Then show that you can reload a backbone, tokenizer, or causal lm object from this directory. We want to help build the user mental model for what a preset is.

Reply:
Good idea! I agree that we need to build the user's mental model. 
What do you think about just adding `os.listdir()` here and creating an extensive section at the end about various ways of loading? Both from local and remote presets.
Something like this:
Sources of preset:
```
causal_lm.from_preset(local_dir)
causal_lm.from_preset(kaggle_handle)
causal_lm.from_preset(hf_handle)
```
Different components that can be loaded from preset:
```
Backbone.from_preset(local_dir)
Tokenizer.from_preset(local_dir)
Preprocessor.from_preset(local_dir)
Classifier.from_preset(local_dir)
causal_lm.from_preset(local_dir)
```

Reply:
I added loading from local dir to here (before upload). 
Also added loading the model from a model hub after upload.